### PR TITLE
Updates Endpoints Frameworks v2 sample with service management.

### PR DESCRIPTION
**Pull request by wmwong**
Adds service management and auth dependencies and filters.
Uses exec-maven-plugin to generate the swagger spec.
Updates README for swagger generation.

### CODE REVIEW

File: appengine/endpoints-frameworks-v2/backend/pom.xml [hunk 1]
```diff
@@ -31,9 +31,21 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
+        <endpoints.framework.version>2.0.0-beta.7</endpoints.framework.version>
```
Comment (lesv) [inline appengine/endpoints-frameworks-v2/backend/pom.xml]: I would hope this changes, but ok.

File: appengine/endpoints-frameworks-v2/backend/pom.xml [hunk 2]
```diff
@@ -31,9 +31,21 @@
...
+        <endpoints.management.version>0.0.20-SNAPSHOT</endpoints.management.version>
```
Comment (lesv) [inline appengine/endpoints-frameworks-v2/backend/pom.xml]: This should change before launch.

File: appengine/endpoints-frameworks-v2/backend/pom.xml [hunk 3]
```diff
@@ -31,9 +31,21 @@
...
+
         <endpoints.project.id>YOUR_PROJECT_ID</endpoints.project.id>
     </properties>
 
+    <!-- TODO(wmwong): Remove this repo before merging. -->
```
Comment (lesv) [inline appengine/endpoints-frameworks-v2/backend/pom.xml]: Yea - and we don't usually put ldap's in public repos.

File: appengine/endpoints-frameworks-v2/backend/README.md [hunk 4]
```diff
@@ -27,16 +27,7 @@ To build the project:
...
-package](http://search.maven.org/remotecontent?filepath=com/google/endpoints/endpoints-framework-tools/2.0.0-beta.7/endpoints-framework-tools-2.0.0-beta.7.zip).
-
-0. Invoke the Endpoints Tool using this command:
-
-         path/to/endpoints-framework-tools-2.0.0-beta.7/bin/endpoints-framework-tools get-swagger-doc \
-             -h <PROJECT_ID>.appspot.com \
-             -w target/echo-1.0-SNAPSHOT com.example.echo.Echo
-
-    Replace`<PROJECT_ID>` with your project ID.
+    mvn exec:java -DGetSwaggerDoc
```
Comment (lesv) [inline appengine/endpoints-frameworks-v2/backend/README.md]: Do you mention what files to edit to make things work somewhere here?

**Comment by lesv** [comment]
LGTM - with required changes.

**Comment by lesv** [comment]
I'm marking LGTM and DO NOT MERGE - until all is ok.

File: appengine/endpoints-frameworks-v2/backend/README.md (hunk 4 above)
Comment (wmwong) [inline appengine/endpoints-frameworks-v2/backend/README.md]: Yes. Up above, there is a section to replace the endpoints.project.id property.

File: appengine/endpoints-frameworks-v2/backend/pom.xml (hunk 2 above)
Comment (wmwong) [inline appengine/endpoints-frameworks-v2/backend/pom.xml]: Changed to 1.0.0-beta.7.

File: appengine/endpoints-frameworks-v2/backend/pom.xml (hunk 3 above)
Comment (wmwong) [inline appengine/endpoints-frameworks-v2/backend/pom.xml]: This entire section has been removed.

**Comment by lesv** [comment]
@tswast FYI - BQ Samples are failing.

**Comment by tswast** [comment]
Merging. LGTM, and it sounds like this is ready from chat.

**Pull request merged**